Frameworks/Foundation/NSUserDefaults.mm

```diff
@@ -119,16 +132,44 @@ - (void)setPersistentDomain:(id)domain forName:(NSString*)name {
  @Status Interoperable
 */
 - (BOOL)synchronize {
-    return CFPreferencesAppSynchronize(kCFPreferencesCurrentApplication);
+    _cacheLock.lock();
```

Comment:
We need fine control over the lock here, as CFPreferencesAppSynchronize should be called outside of _cacheLock.

---
In reply to: [70540876](https://github.com/Microsoft/WinObjC/pull/612#discussion_r70540876) [](ancestors = 70540876)